Stratis.Bitcoin/RPC/Controllers/FullNodeController.cs

```diff
@@ -1,51 +1,132 @@
 ﻿using Microsoft.AspNetCore.Mvc;
-using NBitcoin;
+using Microsoft.Extensions.PlatformAbstractions;
+using Stratis.Bitcoin.Configuration;
 using Stratis.Bitcoin.RPC.Models;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using Stratis.Bitcoin.BlockStore;
+using Stratis.Bitcoin.MemoryPool;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Utilities;
 
 namespace Stratis.Bitcoin.RPC.Controllers
 {
     //TODO: Need to be extensible, should be FullNodeController
-    public partial class ConsensusController : Controller
+    public partial class FullNodeController : Controller
     {
+        private Builder.IFullNode _FullNode;
+        private NodeSettings _Settings;
+        private NBitcoin.Network _Network;
+        private ConsensusValidator _ConsensusValidator;
+        private NBitcoin.ChainBase _Chain;
+        private ChainBehavior.ChainState _ChainState;
+        private BlockStoreManager _BlockManager;
+        private MempoolManager _MempoolManager;
+        private Connection.ConnectionManager _ConnectionManager;
+
+        public FullNodeController(
```

Comment:
This has now been subclassed under BaseRPCController so that all controllers can access whichever services are available, along with guards for what were already required services for those existing controllers.

I'm not sure if there should just be a single controller class or not. Thoughts?